root/statistics/Relationships.js

```diff
@@ -0,0 +1,123 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import {compare} from '../static/scripts/common/i18n';
+import {l_relationships} from '../static/scripts/common/i18n/relationships';
+import {l_statistics as l} from '../static/scripts/common/i18n/statistics';
+import {withCatalystContext} from '../context';
+import manifest from '../static/manifest';
+import formatEntityTypeName from '../static/scripts/common/utility/formatEntityTypeName';
+
+import {formatCount, formatPercentage} from './utilities';
+import StatisticsLayout from './StatisticsLayout';
+
+export type RelationshipsStatsT = {|
+  +$c: CatalystContextT,
+  +dateCollected: string,
+  +stats: {[string]: number},
+  +types: {[string]: RelationshipTypeT},
+|};
+
+declare type RelationshipTypeT = {|
+  +entity_types: $ReadOnlyArray<string>,
+  +tree: {[string]: Array<LinkTypeT>},
+|};
+
+function comparePhrases(a, b) {
+  return compare(a.long_link_phrase, b.long_link_phrase);
+}
+
+const TypeRows = withCatalystContext(({
+  $c,
+  base,
+  indent,
+  parent,
+  stats,
+  type,
+}) => {
+  return (
+    <>
+      <tr>
+        <th style={{paddingLeft: (indent - 1) + 'em'}}>{l_relationships(type.long_link_phrase)}</th>
+        <td>{formatCount($c, stats[base + '.' + type.name])}</td>
+        <td>{formatCount($c, stats[base + '.' + type.name + '.inclusive'])}</td>
+        <td>{formatPercentage($c, stats[base + '.' + type.name + '.inclusive'] / stats[parent], 1)}</td>
+      </tr>
+      {type.children ? (
+        type.children.sort(comparePhrases).map((child) => (
+          <TypeRows base={base} indent={indent + 1} key={child.id} parent={base + '.' + type.name + '.inclusive'} stats={stats} type={child} />
+        ))
+      ) : null}
+    </>
+  );
+});
+
+const Relationships = ({
+  $c,
+  dateCollected,
+  stats,
+  types,
+}: RelationshipsStatsT) => (
+  <StatisticsLayout fullWidth page="relationships" title={l('Relationships')}>
+    {manifest.css('statistics')}
+    <p>
+      {l('Last updated: {date}',
+        {date: dateCollected})}
```

Comment:
Random observation: this could easily fit on the same line as 72, and yet there are a bunch of lines between 102-112 below which extend way beyond 78 chars and desperately need line breaks, lol.